master/test/streamdata/stream_data.go

```diff
@@ -0,0 +1,45 @@
+//go:build integration
+// +build integration
+
+package streamdata
+
+import (
+	// embed is only used in comments.
+	_ "embed"
+	"testing"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/test/migrationutils"
+)
+
+//go:embed stream_trials.sql
+var streamTrialsSQL string
+
+const latestMigration = 20230830174810
+
+// StreamTrialsData holds the migration function and relevant information
+type StreamTrialsData struct {
+	MustMigrate migrationutils.MustMigrateFn
+	ExpID       int
+	JobID       string
+	TaskIDs     []string
+	TrialIDs    []int
+}
+
+// GenerateStreamTrials fills the database with dummy experiment, trials, jobs, and tasks.
+func GenerateStreamTrials() StreamTrialsData {
```

Comment:
question: will we need to also update this whenever we add more streamable types or are we thinking of creating a `GenerateStreamX()` for each type?

Reply:
Yeah I think this will mean adding more streamable types - I think this is about the limit for how generic we can make the testing, but I'll make it as close as possible to being plug-and-play from this point on